src/vsr/replica.zig

```diff
@@ -63,6 +63,7 @@ const CommitStage = enum {
     checkpoint_state_machine,
     checkpoint_client_replies,
     checkpoint_grid_repair,
+    checkpoint_free_set,
```

Comment:
Ahhh, I tried that, an then got a weird error in vopr, and then realized that this doesn't work, at least without some extra legwork. 

`checkpoint_free_set` is what "buffer-flips" the free set right now. So, if we run grid-repair _after_ checkpointing the free set, it'll see the new checkpoint and won't know which blocks are released. 

What we can do is change the logic of repair checkpoint to _assume_ that it happens after buffer flip and use is_free rather than is_released:

https://github.com/tigerbeetle/tigerbeetle/commit/b06bf4a882e85e3b9042790177f20e985b350f54

In some sense, this even feels clearer, as we are waiting the writes to the free blocks, which is easier to see how it could lead to troubles.

Reply:
I think I'll do it, but in a follow up. One thing I don't like is how subtle it is when the data structures "flip" to the next checkpoint, maybe I find some better way to assert that...